examples/semi_supervised/plot_semi_supervised_newsgroups.py

```diff
@@ -72,40 +94,89 @@
 )
 
 
-def eval_and_print_metrics(clf, X_train, y_train, X_test, y_test):
-    print("Number of training samples:", len(X_train))
-    print("Unlabeled samples in training set:", sum(1 for x in y_train if x == -1))
+def eval_and_get_f1(clf, X_train, y_train, X_test, y_test):
+    """Evaluate model performance and return F1 score"""
+    print(f"   Number of training samples: {len(X_train)}")
+    print(f"   Unlabeled samples in training set: {sum(1 for x in y_train if x == -1)}")
     clf.fit(X_train, y_train)
     y_pred = clf.predict(X_test)
-    print(
-        "Micro-averaged F1 score on test set: %0.3f"
-        % f1_score(y_test, y_pred, average="micro")
-    )
-    print("-" * 10)
-    print()
+    f1 = f1_score(y_test, y_pred, average="micro")
+    print(f"   Micro-averaged F1 score on test set: {f1:.3f}")
+    print("\n")
+    return f1
 
 
-if __name__ == "__main__":
-    X, y = data.data, data.target
-    X_train, X_test, y_train, y_test = train_test_split(X, y)
+X, y = data.data, data.target
+X_train, X_test, y_train, y_test = train_test_split(X, y)
 
-    print("Supervised SGDClassifier on 100% of the data:")
-    eval_and_print_metrics(pipeline, X_train, y_train, X_test, y_test)
+f1_scores = {}
 
-    # select a mask of 20% of the train dataset
-    y_mask = np.random.rand(len(y_train)) < 0.2
+# Evaluate supervised model with 100% of training data
```

Comment:
We should probably split each call to `eval_and_get_f1` in its own cell by using `# %%` delimiters. Then the comments such as `# Evaluate supervised model with 100% of training data` should be slightly rephrased as grammatically correct sentences or paragraphs instead of just short inline code comments as @StefanieSenger suggested above.